addons/sale/models/res_partner.py

```diff
@@ -54,3 +54,10 @@ def action_view_sale_order(self):
         all_child = self.with_context(active_test=False).search([('id', 'child_of', self.ids)])
         action["domain"] = [("partner_id", "in", all_child.ids)]
         return action
+
+    def _credit_debit_get(self):
```

Comment:
`_get_additional_amount()` is meant to make sure the `updated_credit` does not compute the the same amount twice.
Indeed, when creating an invoice from a SO, we don't want to include the current invoice in `updated_credit` since it is already included in the partner's `credit`.
But if, for some reason, the user decides to change the invoice amount and make it larger than the `amount_to_invoice`, then, we want to add the exceeding amount to the `update_credit` :wink:

Reply:
I'm not sure I follow you, please add a test for that use case

Reply:
Done :heavy_check_mark: